common/djangoapps/student/views.py

```diff
@@ -357,24 +357,29 @@ def signin_user(request):
     external_auth_response = external_auth_login(request)
     if external_auth_response is not None:
         return external_auth_response
+    # Determine the URL to redirect to following login:
+    redirect_to = get_next_url_for_login_page(request)
     if request.user.is_authenticated():
-        return redirect(reverse('dashboard'))
+        return redirect(redirect_to)
+
+    third_party_auth_error = None
+    for msg in messages.get_messages(request):
+        if msg.extra_tags.split()[0] == "social-auth":
+            third_party_auth_error = unicode(msg)
```

Comment:
Potentially, yes. The template should always prefix the error with a translated message like "Sorry, that login didn't work:" and then this technical explanation which may not be translated. I guess I'm assuming that giving a specific reason, even if untranslated, along with a translated generic error will be more helpful than not.